src/write.rs

```diff
@@ -336,38 +336,12 @@ where
         self
     }
 
-    /// Gives the sequence of operations that have not yet been applied.
-    ///
-    /// Note that until the *first* call to `refresh`, the sequence of operations is always empty.
-    ///
-    /// ```
-    /// # use evmap::Operation;
-    /// let x = ('x', 42);
-    ///
-    /// let (r, mut w) = evmap::new();
-    ///
-    /// // before the first refresh, no oplog is kept
-    /// w.refresh();
-    ///
-    /// assert_eq!(w.pending(), &[]);
-    /// w.insert(x.0, x);
-    /// assert_eq!(w.pending(), &[Operation::Add(x.0, x)]);
-    /// w.refresh();
-    /// w.remove(x.0, x);
-    /// assert_eq!(w.pending(), &[Operation::Remove(x.0, x)]);
-    /// w.refresh();
-    /// assert_eq!(w.pending(), &[]);
-    /// ```
-    pub fn pending(&self) -> &[Operation<K, V>] {
-        &self.oplog[self.swap_index..]
-    }
-
     /// Refresh as necessary to ensure that all operations are visible to readers.
     ///
     /// `WriteHandle::refresh` will *always* wait for old readers to depart and swap the maps.
     /// This method will only do so if there are pending operations.
     pub fn flush(&mut self) -> &mut Self {
-        if !self.pending().is_empty() {
+        if !self.oplog[self.swap_index..].is_empty() {
```

Comment:
Now that we don't have `pending()` for the added readability, I kind of want to just check
```suggestion
        if self.swap_index < self.oplog.len() {
```
or whatever makes this work 🤔

Reply:
Yes, that should be right. Running the test suite should confirm it pretty quickly.

Reply:
I like the confidence! I wasn't sure because it looked like [only one test called `flush` and it didn't test much](https://github.com/jonhoo/rust-evmap/blob/master/tests/lib.rs#L189) but I'll go ahead with that change

Reply:
I guess I should maybe update/write some test to ensure that `refresh` isn't called if I call `flush` twice or something? Is that a thing that is possible to verify?

Reply:
Huh, interesting. I thought `flush` was called internally, but I guess not. Yes, then adding a test that checks that would be useful indeed! You'll probably need to write it as a unit test and then inspect some internal state to check if `refresh` was called.

Reply:
Added 351881d37f68326de12618ff8606969e76d1a876. I'm not sure if `w.second` is the best idea but there wasn't anything _terribly_ obvious